test/library/standard/DateTime/strptime.bad

```diff
@@ -0,0 +1,2 @@
+test_microsecond()
+strptime.chpl:7: error: halt reached - day is out of the valid range
```

Comment:
Calling this method with an undefined format specifier is resulting in undefined behavior. We could potentially add verification if this is a problem we think is important.

Reply:
OK, thanks for the diagnosis David.  When you say verification, are you thinking that we would validate the entries returned by the C routine before passing them to the `datetime` initializer?  (I don't think we can validate the format string itself without creating our own parser for it which seems to be challenging...?)  I don't personally feel a need to do this, at least until we get indications that users are using the routine, though it's a shame to have this behavior today.

Reply:
I was thinking we'd need to validate the format string, but looking closer we could probably just validate the entries returned by the C function to at least ensure they're in a reasonable range. We may still get unlucky and have garbage that is in the valid range though.

Reply:
Could we fill the struct with known-garbage values prior to sending it to C and then checking for validity of all fields before passing it to the initializer?  (C only writes, never reads, the fields, is that right?)

Reply:
Yes, that sounds reasonable.